genie-web/src/test/java/com/netflix/genie/web/configs/GenieAgentFilterAutoConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.configs;
+
+import com.netflix.genie.test.categories.IntegrationTest;
+import com.netflix.genie.web.services.AgentFilterService;
+import com.netflix.genie.web.util.AgentMetadataInspector;
+import com.netflix.genie.web.util.BlacklistedVersionAgentMetadataInspector;
+import com.netflix.genie.web.util.MinimumVersionAgentMetadataInspector;
+import com.netflix.genie.web.util.WhitelistedVersionAgentMetadataInspector;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+
+/**
+ * Tests for behavior of {@link GenieAgentFilterAutoConfiguration}.
+ *
+ * @author tgianos
+ * @since 4.0.0
+ */
+@Category(IntegrationTest.class)
+public class GenieAgentFilterAutoConfigurationIntegrationTest {
+
+    /**
+     * Test expected context when configuration is enabled.
+     */
+    @Test
+    public void testContextIfEnabled() {
+        new ApplicationContextRunner()
```

Comment:
Just an FYI you can make the part with the `AutoConfigurations.of` common member of the class to reuse. It all acts as a builder so anything you add/modify within each test method will build on the common. It looks like this part is common to all the tests where you set only this auto configuration